components/TaskCard.js

```diff
@@ -34,57 +37,56 @@ const StyledTaskCard = styled.div`
             width: 70px;
         }
         
         .Medium {
-            background-color: rgba(255, 196, 0, 0.75);
+            background-color: var(--bg-medium);
             width: 70px;
         }
         
         .Low {
-            background-color: rgba(21, 255, 0, 0.75);
+            background-color: var(--bg-low);
             width: 70px;
         }
 
         button {
             background-color: white;
-            border: 1px solid black;
-            border-radius: var(--border-radius-button);
+
+            display: flex;
+            align-items: center;
+            justify-content: center;
         }
 
         svg {
             height: 20px;
         }
-    `}
+
+        ${props => props.$isDone ? `background-color: lightgrey; color: grey; text-decoration:line-through; ` : ``}
+  `};
+`;
+
+const StyledToggebuttonWrapper = styled.div`
+    width: 75px;
+    button {
+        border: 1px solid black;
+        border-radius: 5px;
+    }
 `;
 
-export default function TaskCard({ task, onDeleteTask }) {
+export default function TaskCard({ task, onDeleteTask, toggleDone}) {
     const [isDeleteOption, setIsDeleteOption] = useState(false);
     const [toggleButtonName, setToggleButtonName] = useState("Delete");
     const router = useRouter();
 
-    // SVG s
-
-    const MagnifyingGlassIcon = () => {
-        return (
-            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
-            {/* <!--!Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2024 Fonticons, Inc.--> */}
-            <path d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"/></svg>
-        );
-      };
-    const Trashcan = () => {
-        return (
-            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512">
-            {/* <!--!Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2024 Fonticons, Inc.--> */}
-            <path d="M170.5 51.6L151.5 80l145 0-19-28.4c-1.5-2.2-4-3.6-6.7-3.6l-93.7 0c-2.7 0-5.2 1.3-6.7 3.6zm147-26.6L354.2 80 368 80l48 0 8 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-8 0 0 304c0 44.2-35.8 80-80 80l-224 0c-44.2 0-80-35.8-80-80l0-304-8 0c-13.3 0-24-10.7-24-24S10.7 80 24 80l8 0 48 0 13.8 0 36.7-55.1C140.9 9.4 158.4 0 177.1 0l93.7 0c18.7 0 36.2 9.4 46.6 24.9zM80 128l0 304c0 17.7 14.3 32 32 32l224 0c17.7 0 32-14.3 32-32l0-304L80 128zm80 64l0 208c0 8.8-7.2 16-16 16s-16-7.2-16-16l0-208c0-8.8 7.2-16 16-16s16 7.2 16 16zm80 0l0 208c0 8.8-7.2 16-16 16s-16-7.2-16-16l0-208c0-8.8 7.2-16 16-16s16 7.2 16 16zm80 0l0 208c0 8.8-7.2 16-16 16s-16-7.2-16-16l0-208c0-8.8 7.2-16 16-16s16 7.2 16 16z"/></svg>
-        );
-      };
-
     function getVariant(task) {
```

Comment:
- You can simplify  the `getVariant` function
- Then, you can handle the conditional styling directly within `StyledTaskCard` using styled-components.

_Example:_

```js

const StyledTaskCard = styled.div`
// other code 
 
  background-color: ${({ $variant }) =>
    $variant === "overdue"
      ? "var(--bg-overdue)"
      : $variant === "today"
      ? "var(--bg-today)"
      : "var(--bg-default)"};
`

// other code 

//  Determine the variant
  function getTaskVariant(dueDate) {
    const currentDate = new Date().toISOString().split("T")[0];
    const taskDueDate = new Date(dueDate).toISOString().split("T")[0];

    if (currentDate > taskDueDate) return "overdue";
    if (currentDate === taskDueDate) return "today";
    return "default";
  }
  
  // get the Variant
  const taskVariant = getTaskVariant(task.dueDate);
  
  
  return (
    <StyledTaskCard $variant={taskVariant} $isDone={task.isDone}>
  // other code
  )
  

```

Reply:
thank you 🌷